enterprise/api_client/lms.py

```diff
@@ -434,8 +442,24 @@ def get_course_assessment_grades(self, course_id, username):
         * ``percent``: A float representing the overall grade for the course.
         * ``module_id``: The ID of the subsection.
         """
-        api_url = self.get_api_url(f"gradebook/{course_id}")
-        response = self.client.get(api_url, params={"user_contains": username})
+        attempts = 0
+        while True:
+            attempts = attempts + 1
+            api_url = self.get_api_url(f"gradebook/{course_id}")
+            try:
+                response = self.client.get(api_url, params={"user_contains": username}, timeout=40)
+            except Timeout as to_exception:
+                if attempts <= self.MAX_RETRIES:
+                    sleep_seconds = self._calculate_backoff(attempts)
+                    LOGGER.warning(
+                        f"[ATTEMPT: {attempts}] Request to the LMS grades API timeouted out with "
+                        f"exception: {to_exception}, backing off for {sleep_seconds} seconds and retrying"
+                    )
+                    time.sleep(sleep_seconds)
```

Comment:
Question:  Is the reason we're doing a separate sleep, rather than just retrying the get request with a longer timeout, to lighten the load on the service being called?

Reply:
yup! We want to lesson the load on the grades api